theseus/core/objective.py

```diff
@@ -285,7 +285,7 @@ def _add_batched_cost_function(
         else:
             if batch_name not in self.batched_cost_functions:
                 cost_function_batch = cost_function.copy(
-                    batch_name + "__cost_function_batch"
+                    batch_name + "__cost_function_batch", keep_variable_names=True
```

Comment:
Is there a reason to keep variable names? Just curious, it looks like there should be no problems with this.

Reply:
I was running into errors when populating the batched cost function because `AutoDiffCostFunction` would create attributes named like `"obj_pose_0"`, but the copy would have attribute names like `"obj_pose_0_copy"`.